applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h

```diff
@@ -752,19 +724,19 @@ class GearScheme : public Scheme<TSparseSpace, TDenseSpace>
             this->PeriodicConditionProjectionCorrection(rModelPart);
         }
 
-        for (typename ModelPart::NodesContainerType::iterator iNode = rModelPart.NodesBegin(); iNode != rModelPart.NodesEnd(); iNode++)
-        {
-            if (iNode->FastGetSolutionStepValue(NODAL_AREA) == 0.0)
-            {
+        const double zero_tol = 1.0e-12;
+#pragma omp parallel for firstprivate(zero_tol)
+        for (int i_node = 0; i_node < n_nodes; ++i_node){
+            auto iNode = rModelPart.NodesBegin() + i_node;
+            if (iNode->FastGetSolutionStepValue(NODAL_AREA) < zero_tol) {
                 iNode->FastGetSolutionStepValue(NODAL_AREA) = 1.0;
             }
             const double Area = iNode->FastGetSolutionStepValue(NODAL_AREA);
             iNode->FastGetSolutionStepValue(ADVPROJ) /= Area;
             iNode->FastGetSolutionStepValue(DIVPROJ) /= Area;
         }
 
-        KRATOS_INFO_IF("GearScheme", rModelPart.GetCommunicator().MyPID() == 0)
-            << "Computing OSS projections" << std::endl;
+        KRATOS_INFO("BDF2TurbulentScheme") << "Computing OSS projections" << std::endl;
```

Comment:
This is very verbose, can you protect it with an `echo_level`?

Reply:
OK.

Reply:
I checked the base class and there is no echo level in the  `Scheme`. Probably we should add it in there before.